src/TLuaInterpreter.cpp

```diff
@@ -15198,55 +15215,87 @@ void TLuaInterpreter::initIndenterGlobals()
+    };
+    QStringList failedMessages{};
 
-    int error = luaL_dofile(pGlobalLua, path.toUtf8().constData());
-    if (error != 0) {
-        // For the installer we do not go down a level to search for this. So
-        // we check again for the user case of a windows install.
 
-        // overload previous behaviour to check by absolute path as well
-        // TODO this sould be cleaned up and refactored to just use an array and a for loop
-        path = QCoreApplication::applicationDirPath() + "/mudlet-lua/lua/LuaGlobal.lua";
+    int error;
+    for (const auto& path : qAsConst(pathsToTry)) {
         if (!QFileInfo::exists(path)) {
-            path = "mudlet-lua/lua/LuaGlobal.lua";
+            failedMessages << tr("%1 (doesn't exist)", "This file doesn't exist").arg(path);
+            continue;
+        }
+
+        // load via Qt so UTF8 paths work on Windows - Lua can't handle it
+        auto luaGlobal = readScriptFile(path);
+        if (luaGlobal.isEmpty()) {
+            failedMessages << tr("%1 (couldn't read file)", "This file could not be read for some reason (for example, no permission)").arg(path);
+            continue;
         }
-        error = luaL_dofile(pGlobalLua, path.toUtf8().constData());
+
+        error = luaL_dostring(pGlobalLua, luaGlobal.toUtf8().constData());
         if (error == 0) {
             mpHost->postMessage(tr("[  OK  ]  - Mudlet-lua API & Geyser Layout manager loaded."));
             return;
+        } else {
+            qWarning() << "TLuaInterpreter::loadGlobal() loading " << path << " failed: " << lua_tostring(pGlobalLua, -1);
+            failedMessages << QStringLiteral("%1 (%2)").arg(path, lua_tostring(pGlobalLua, -1));
         }
-    } else {
-        mpHost->postMessage(tr("[  OK  ]  - Mudlet-lua API & Geyser Layout manager loaded."));
+    }
+
+    mpHost->postMessage(tr("[ ERROR ] - Couldn't to load LuaGlobal; your Mudlet is broken! Tried these locations:\n%1").arg(failedMessages.join(QStringLiteral("\n"))));
+}
+
+// No documentation available in wiki - internal function
+// Returns contents of the file or empty string if it couldn't be read
+QString TLuaInterpreter::readScriptFile(const QString& path) const
+{
+    QFile file(path);
+    if (!file.open(QFile::ReadOnly | QFile::Text)) {
+        return QString();
+    }
+
+    QTextStream in(&file);
+    QString text = in.readAll();
+    file.close();
+
+    return text;
+}
+
+#if defined(Q_OS_WIN32)
+// No documentation available in wiki - internal function
+// loads utf8_filenames from the resource system directly so it is not affected by
+// non-ASCII characters that might be present in the users filesystem
+void TLuaInterpreter::loadUtf8Filenames()
+{
+    auto path = QStringLiteral(":/mudlet-lua/lua/utf8_filenames.lua");
+    auto text = readScriptFile(path);
+    if (text.isEmpty()) {
+        qWarning() << "TLuaInterpreter::loadUtf8Filenames() ERROR: couldn't file for reading: " << path;
```

Comment:
`qWarning() << "TLuaInterpreter::loadUtf8Filenames() ERROR: couldn't file for reading: " << path;`

"...Couldn't {WTF to} file for reading..." - not sure the above makes sense :wink: